go/enclave/enclave.go

```diff
@@ -600,7 +601,62 @@ func (e *enclaveImpl) EstimateGas(encryptedParams common.EncryptedParamsEstimate
 }
 
 func (e *enclaveImpl) GetLogs(encryptedParams common.EncryptedParamsGetLogs) (common.EncryptedResponseGetLogs, error) {
-	return nil, fmt.Errorf("eth_getLogs is not implemented")
+	// We decrypt the params.
+	paramBytes, err := e.rpcEncryptionManager.DecryptBytes(encryptedParams)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decrypt params in GetLogs request. Cause: %w", err)
+	}
+
+	// We verify the params.
+	var paramsList []interface{}
```

Comment:
Can we move all this decrypting/deserializing/extracting somewhere else for readability, maybe get it down to something like
`forAddress, filter, err := extractFilter(encryptedParams)`
or whatever similar methods are doing.